lib/compiler_rt/hexagon.zig

```diff
@@ -0,0 +1,1852 @@
+        \\   r5:4 = sub(r1:0,r9:8,p2):carry
+        \\
+        \\
+        \\
+        \\
+        \\   r7:6 = #1
+        \\   r11:10 = #0
+        \\  }
+        \\  {
+        \\   r3:2 = sub(r15:14,r11:10,p2):carry
+        \\   r7:6 = add(r13:12,r7:6)
+        \\   r28 = add(r28,#-0x3ff)
+        \\  }
+        \\  {
+        \\
+        \\   if (p2) r13:12 = r7:6
+        \\   if (p2) r1:0 = r5:4
+        \\   if (p2) r15:14 = r3:2
+        \\  }
+        \\  {
+        \\   r5:4 = sub(r1:0,r9:8,p3):carry
+        \\   r7:6 = #1
+        \\   r28 = asr(r28,#1)
+        \\  }
+        \\  {
+        \\   r3:2 = sub(r15:14,r11:10,p3):carry
+        \\   r7:6 = add(r13:12,r7:6)
+        \\  }
+        \\  {
+        \\   if (p3) r13:12 = r7:6
+        \\   if (p3) r1:0 = r5:4
+        \\
+        \\
+        \\
+        \\
+        \\
+        \\   r2 = #1
+        \\  }
+        \\  {
+        \\   p0 = cmp.eq(r1:0,r11:10)
+        \\   if (!p0.new) r12 = or(r12,r2)
+        \\   r3 = cl0(r13:12)
+        \\   r28 = add(r28,#-63)
+        \\  }
+        \\
+        \\
+        \\
+        \\  {
+        \\   r1:0 = convert_ud2df(r13:12)
+        \\   r28 = add(r28,r3)
+        \\  }
+        \\  {
+        \\   r1 += asl(r28,#52 -32)
+        \\   jumpr r31
+        \\  }
+        \\ .Lsqrt_abnormal:
+        \\  {
+        \\   p0 = dfclass(r1:0,#0x01)
+        \\   if (p0.new) jumpr:t r31
+        \\  }
+        \\  {
+        \\   p0 = dfclass(r1:0,#0x10)
+        \\   if (p0.new) jump:nt .Lsqrt_nan
+        \\  }
+        \\  {
+        \\   p0 = cmp.gt(r1,#-1)
+        \\   if (!p0.new) jump:nt .Lsqrt_invalid_neg
+        \\   if (!p0.new) r28 = ##0x7F800001
+        \\  }
+        \\  {
+        \\   p0 = dfclass(r1:0,#0x08)
+        \\   if (p0.new) jumpr:nt r31
+        \\  }
+        \\
+        \\
+        \\  {
+        \\   r1:0 = extractu(r1:0,#52,#0)
+        \\  }
+        \\  {
+        \\   r28 = add(clb(r1:0),#-11)
+        \\  }
+        \\  {
+        \\   r1:0 = asl(r1:0,r28)
+        \\   r28 = sub(#1,r28)
+        \\  }
+        \\  {
+        \\   r1 = insert(r28,#1,#52 -32)
+        \\  }
+        \\  {
+        \\   r3:2 = extractu(r1:0,#23 +1,#52 -23)
+        \\   r5 = ##0x3f000004
+        \\  }
+        \\  {
+        \\   r9 = or(r5,r2)
+        \\   r5 = and(r5,#-16)
+        \\   jump .Ldenormal_restart
+        \\  }
+        \\ .Lsqrt_nan:
+        \\  {
+        \\   r28 = convert_df2sf(r1:0)
+        \\   r1:0 = #-1
+        \\   jumpr r31
+        \\  }
+        \\ .Lsqrt_invalid_neg:
+        \\  {
+        \\   r1:0 = convert_sf2df(r28)
+        \\   jumpr r31
+        \\  }
+    );
+}
+
+comptime {
+    if (builtin.cpu.arch == .hexagon) {
+        @export(__hexagon_adddf3, .{ .name = "__hexagon_adddf3", .linkage = common.linkage, .visibility = common.visibility });
+        @export(__hexagon_adddf3, .{ .name = "__qdsp_adddf3", .linkage = common.linkage, .visibility = common.visibility });
```

Comment:
From what I can see, LLVM doesn't use the `__qdsp_`-prefixed names. Perhaps this was for an older compiler?